### PR TITLE
genai: code execution

### PR DESCRIPTION
**Pull request by jba**
Add the new parts needed for code execution.

### CODE REVIEW

File: genai/example_test.go [hunk 1]
```diff
@@ -101,6 +101,29 @@ func ExampleGenerativeModel_GenerateContent_safetySetting() {
...
 
+func ExampleGenerativeModel_GenerateContent_codeExecution() {
+	ctx := context.Background()
+	client, err := genai.NewClient(ctx, option.WithAPIKey(os.Getenv("GEMINI_API_KEY")))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	model := client.GenerativeModel("gemini-1.0-pro")
```
Comment (eliben) [inline genai/example_test.go:112]: Nit: let's use newer models for new examples; I think there was a request to suggest using new models in docs/examples whenever possible
you're using 1.5-flash for the test...

File: go.mod [hunk 2]
```diff
@@ -3,19 +3,19 @@ module github.com/google/generative-ai-go
 go 1.21
 
 require (
-	cloud.google.com/go/ai v0.7.0
+	cloud.google.com/go/ai v0.7.1-0.20240627174219-eec7a3b5c00f
```
Comment (eliben) [inline go.mod:6]: https://github.com/googleapis/google-cloud-go/pull/10452 should be ready to land soon so we should be able to depend on an actually released version here

Comment (jba) [reply go.mod:6]: I just merged it.

File: genai/example_test.go (hunk 1 above)
Comment (jba) [reply genai/example_test.go:112]: Done here and elsewhere in the examples.

File: go.mod (hunk 2 above)
Comment (jba) [reply go.mod:6]: I'm having trouble `go get`ing v0.80. Maybe it just needs a little time.

Comment (eliben) [reply go.mod:6]: It's tagged now https://github.com/googleapis/google-cloud-go/releases/tag/ai%2Fv0.8.0

Comment (jba) [reply go.mod]: Done.

**Review by eliben — APPROVED** [review]

**Pull request merged**